packages/components/src/Interactive.ts

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash/fp';
+import { createElement, forwardRef, ReactNode, useState } from 'react';
+
+import * as Utils from './utils';
+
+const allowedHoverVariables = [
+  'background',
+  'backgroundColor',
+  'border',
+  'color',
+  'boxShadow',
+  'opacity',
+  'textDecoration',
+];
+const pointerTags = ['button', 'area', 'a', 'select'];
+const pointerTypes = ['radio', 'checkbox', 'submit', 'button'];
+
+export type InteractiveProps = {
+  as: keyof JSX.IntrinsicElements;
+  children?: ReactNode;
+  className?: string;
+  disabled?: boolean;
+  hover?: React.CSSProperties;
+  role?: string;
+  style?: React.CSSProperties;
+  tabIndex?: number;
+  type?: string;
+  onBlur?: (event: any) => void;
+  onClick?: (event: any) => void;
+  onKeyDown?: (event: any) => void;
+  onMouseDown?: (event: any) => void;
+};
+
+export const Interactive: React.ForwardRefExoticComponent<InteractiveProps> = forwardRef(
+  (
+    {
+      as: TagName = 'div',
+      children,
+      className = '',
+      disabled,
+      hover = {},
+      role,
+      style = {},
+      tabIndex,
+      type,
+      onBlur,
+      onClick,
+      onKeyDown,
+      onMouseDown,
+      ...props
+    },
+    ref
+  ) => {
+    const [outline, setOutline] = useState<string>();
+    const { cursor } = style;
+
+    const computedCursor = Utils.cond(
+      [!!cursor, () => cursor],
+      [disabled, () => 'not-allowed'],
+      [!!onClick || pointerTags.includes(TagName) || pointerTypes.includes(type!), () => 'pointer']
+    );
+
+    const computedTabIndex = Utils.cond(
+      [_.isNumber(tabIndex), () => tabIndex],
+      [disabled, () => -1],
+      [!!onClick, () => 0],
+      () => undefined
+    );
+
+    const computedRole = Utils.cond(
+      [!!role, () => role],
+      [onClick && !['input', ...pointerTags].includes(TagName), () => 'button'],
+      () => undefined
+    );
+
+    const cssVariables = _.flow(
+      _.toPairs,
+      _.flatMap(([key, value]) => {
+        console.assert(
+          allowedHoverVariables.includes(key),
+          `${key} needs to be added to the hover-style in style.css for the style to be applied`
+        );
+        return [
+          [`--app-hover-${key}`, value],
+          [key, `var(--hover-${key}, ${style[key]})`],
+        ];
+      }),
+      _.fromPairs
+    )(hover);
+
+    return createElement(
```

Comment:
Yeah, I suppose so. This was copied from All of Us, which doesn't use react-hyperscript-helpers.

Reply:
Yeah, given that it is in the Terra-UI repo, I would rather lean towards Terra-UI practices.

Reply:
Actually, there's nowhere else that uses a dynamic tag name, we always use `div(...` instead of `h('div', ...` etc..

Thus, I never implemented support for that in our type declarations for react-hyperscript-helpers. So using react-hyperscript-helpers here would either require a fair bit of work to update the type declarations or a `ts-expect-error`. I'm inclined to keep `createElement` vs either of those.